deepspeed/utils/timer.py

```diff
@@ -167,62 +171,71 @@ def __init__(
 
     def update_epoch_count(self):
         self.epoch_count += 1
-        self.local_step_count = 0
+        self.local_microstep_count = 0
 
     def _init_timer(self):
         self.initialized = True
 
     def start(self):
         self._init_timer()
         self.started = True
-        if self.total_step_count >= self.start_step:
+        if self.local_microstep_count >= self.start_step * self.gradient_accumulation_steps:
             torch.cuda.synchronize()
             self.start_time = time.time()
 
     def stop(self, report_speed=True):
         if not self.started:
             return
         self.started = False
-        self.total_step_count += 1
-        self.local_step_count += 1
-        if self.total_step_count > self.start_step:
+        self.local_microstep_count += 1
+        gradient_accumulation_boundary = (self.local_microstep_count %
+                                          self.gradient_accumulation_steps == 0)
+        if gradient_accumulation_boundary:
+            self.total_step_count += 1
+
+        if self.local_microstep_count > self.start_step * self.gradient_accumulation_steps:
             torch.cuda.synchronize()
             self.end_time = time.time()
             duration = self.end_time - self.start_time
             self.total_elapsed_time += duration
-
-            curr_samples_sec = (self.batch_size * self.num_workers) / duration
-
-            if report_speed:
-                self.logging(
-                    "{}/{}, RunningAvgSamplesPerSec={}, CurrSamplesPerSec={}, MemAllocated={}GB, MaxMemAllocated={}GB"
-                    .format(
-                        self.epoch_count,
-                        self.local_step_count,
-                        self.avg_samples_per_sec(),
-                        curr_samples_sec,
-                        round(torch.cuda.memory_allocated() / 1024**3,
-                              2),
-                        round(torch.cuda.max_memory_allocated() / 1024**3,
-                              2),
-                    ))
-                if self.monitor_memory:
-                    virt_mem = psutil.virtual_memory()
-                    swap = psutil.swap_memory()
-                    self.logging("{}/{}, vm percent: {}, swap percent: {}".format(
-                        self.epoch_count,
-                        self.local_step_count,
-                        virt_mem.percent,
-                        swap.percent,
-                    ))
+            self.step_elapsed_time += duration
+
+            if gradient_accumulation_boundary:
```

Comment:
```python
   if global_step:
     self.global_step_count += 1
     if report_speed:
         self.logging.... 
```

Reply:
Good point. This logic is one of the oldest in DeepSpeed, and I feel our understanding of useful logging has changed a lot since then, so okay to break this. So how about we log 3 values
1. `self.micro_step_count`: number of micro steps of current epoch, will be reset every epoch
2. `self.global_step_count`: number of global steps of current epoch. Reset every epoch, I think?
3. `self.epoch_count`: number of epochs

What do you think?

Reply:
Sounds good to me, do you have any particular format in mind? `{epoch}/{micro_step}/{global_step}` or `epoch={}/micro_step={}/global_step={}`